LoadBalancer IPAM (#1788)
tigera/docs: calico/reference/resources/ippool.mdx

| nodeSelector                       | Selects the nodes that $[prodname] IPAM should assign addresses from this pool to.                                                                                                  |                                                                                                                                                                                                                        | [selector](#node-selector)            | all()                                         |
| allowedUses _(since v3.21.0)_      | Controls whether the pool will be used for automatic assignments of certain types. See [below](#allowed-uses).                                                                                | Workload, Tunnel, LoadBalancer                                                                                                                                                                                         | list of strings            | `["Workload", "Tunnel"]`                      |

> | assignmentMode                     | Controls whether the pool will be used for automatic assignments or only if requested manualy                                                                                       | Automatic, Manual                                                                                                                                                                                                      | strings                    | `Automatic`                                   |

Reviewer: If this
```suggestion
| assignmentMode                     | Controls whether the pool will be used for automatic assignments or only if requested manually.                                                                                       | Automatic, Manual                                                                                                                                                                                                      | strings                    | `Automatic`                                   |
```
If this is obvious, then ignore. After reading this description, I'm left wondering how I can manually request this pool.

When automatically assigning IP addresses to workloads, only pools with "Workload" in their `allowedUses` field are
consulted. Similarly, when assigning IPs for tunnel devices, only "Tunnel" pools are eligible.

> LoadBalancer option for `allowedUses` field cannot be used in conjunction with either Tunnel or Workload.

Reviewer: ```suggestion
Combining options for the `allowedUses` field is limited. You can specify only the following options and option combinations:
* `allowedUses: ["Tunnel","Workload"]` (default)
* `allowedUses: ["Tunnel"]`
* `allowedUses: ["Workload"]`
* `allowedUses: ["LoadBalancer"]`
```

Reviewer: Aiming for clarity here.

Reducing the block size from the default (e.g., using `28` for IPv4 to give 16 addresses per block) means more blocks per host and therefore potentially more routes. This can be beneficial if it allows the blocks to be more fairly distributed amongst the hosts.

### Assignment Mode

> Determines if the IP Pool should be used by Calico IPAM for automatic IP assignments. With `Automatic` assignmentMode Calico IPAM will automatically assign IP addresses from this pool. By setting assignmentMode to `Manual` Calico IPAM will not assign IP addresses from this pool unless specified by the user.

Reviewer: ```suggestion
Determines if the IP pool should be used by Calico IPAM for automatic IP assignments. With `Automatic` assignmentMode, Calico IPAM automatically assigns IP addresses from this pool. If you set `assignmentMode` to `Manual`, Calico IPAM does not assign IP addresses from this pool unless the user specifies this pool.
```
* Would be good to end the last sentence explaining how/when that happens.
